test/hotspot/jtreg/compiler/sharedstubs/SharedTrampolineTest.java

```diff
@@ -49,6 +50,7 @@ public class SharedTrampolineTest {
     private static void runTest(String test) throws Exception {
         String testClassName = SharedTrampolineTest.class.getName() + "$" + test;
         ArrayList<String> command = new ArrayList<String>();
+        command.add("-XX:-TieredCompilation");  // C2-compiler only
```

Comment:
It is not quite clear to me why we should not pass `args[0]` here, like we do in another test fix. 
Would this test ever make sense to run with different compiler, e.g. C1?

Reply:
Superficially, it looks like the test is optimised for C2. But let's ask @eastig for help.

Reply:
I agree with Aleksey we should use `args[0]`. It looks like we will need to add a test for C1.

Reply:
> I agree with Aleksey we should use args[0]. It looks like we will need to add a test for C1.

Got it, thank you. Fixed.

Reply:
Are we adding C1 version in this PR as well, or in a separate one?

Reply:
I think it is better to add a new test with a separate PR.

Reply:
Created https://bugs.openjdk.org/browse/JDK-8318681 to add tests for C1.